gordo_components/builder/azure_utils.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Iterator, Tuple, Union
+from datetime import datetime
+from pytz import UTC
+import logging
+
+from mlflow.entities import Metric, Param
+from mlflow.tracking import MlflowClient
+from azureml.core import Workspace
+from azureml.core.authentication import (
+    InteractiveLoginAuthentication,
+    ServicePrincipalAuthentication,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_mlflow_client(azureml_config: dict = {}) -> MlflowClient:
+    """
+    Set remote tracking URI for mlflow to AzureML workspace
+
+    Parameters
+    ----------
+    azure_config: dict
+        AzureML Workspace configuration to use for remote MLFlow tracking. Passing only
+        Workspace parameters with a missing our empty `auth` key will result in
+        interactive authentication to AzureML.
+
+        Example::
+
+            `{
+                 "subscription_id":<value>,
+                 "resource_group":<value>,
+                 "workspace_name":<value>
+                 "auth": {
```

Comment:
Think it's more understandable if this is a separate parameter `service_principal_auth_kwargs` with a reference that this is actually kwargs to be supplied to `ServicePrincipalAuthentication` 

And the others are `Workspace` kwargs, no? (less the injection of 'auth' if such kwargs are supplied of course)

Reply:
Agreed :+1: 

Right, those 4 are Workspace args, where `auth` is either an `InteractiveLoginAuthentication` or `ServicePrincipalAuthentication` instance, the latter requiring the kwars shown in the `auth` example.